src/pages/index.js

```diff
@@ -109,43 +114,119 @@ addNewCardButton.addEventListener("click", () => {
 });
 
 function handleAddCardFormSubmit(data) {
-  const name = data.title;
-  const link = data.url;
-  renderCard({ name, link }, cardListEl);
-  newCardModal.close();
+  const name = data.name;
+  const link = data.link;
+  api
+    .newCardReq(name, link)
+    .then((result) => {
+      const { name, link } = result;
+      renderCard(result);
+      newCardModal.close();
+      addCardFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 newCardModal.setEventListeners();
 
+// Card Delete Modal Popup
+
+function handleCardDeleteClick(card) {
+  confirmationModal.open(card);
+}
+
 // --------------------------------------
 
 const editFormModal = new ModalWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 
-
-
 profileEditButton.addEventListener("click", () => {
-  const { name, job } = userInfo.getUserInfo();
+  const { name, about } = userInfo.getUserInfo();
   profileEditForm.querySelector(".modal__input_type_title").value = name;
-  profileEditForm.querySelector(".modal__input_type_description").value = job;
+  profileEditForm.querySelector(".modal__input_type_description").value = about;
   editFormValidator.toggleButtonState();
   editFormModal.open();
 });
 
 function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo({
-    name: data.title,
-    job: data.subtitle,
-  });
-  editFormModal.close();
+  api
+    .uploadProfileReq(data)
+    .then((result) => {
+      userInfo.setUserInfo(result);
+      editFormModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
 editFormModal.setEventListeners();
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
-  jobSelector: ".profile__subtitle",
+  aboutSelector: ".profile__subtitle",
+  avatarSelector: ".profile__image",
+});
+
+// Profile Picture Modal
+
+const profilePictureModal = new ModalWithForm(
+  "#profile-picture-modal",
+  handleProfilePictureSubmit
+);
+
+profilePictureModal.setEventListeners();
+
+profilePictureButton.addEventListener("click", () => {
+  profilePictureModal.open();
 });
 
-// userInfo.setUserInfo({ name: "Steve Jobs", job: "Business Man" });
+function handleProfilePictureSubmit(data) {
+  profilePictureModal.setButtonText(true);
+  api
+    .profilePictureReq(data.link)
+    .then((result) => {
+      profilePictureModal.setButtonText(false);
```

Comment:
` profilePictureModal.setButtonText(false);` should be in `finally`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.